types/index.d.ts

```diff
@@ -0,0 +1,496 @@
+
+    inc(): number;
+
+    /**  Returns the next unique identifier. */
+    fetch(): string;
+
+    /** Resets the identity manager, marking all unique identifiers as available. */
+    reset(): void;
+  }
+}
+
+declare module "miragejs/orm/schema" {
+  import { Collection, ModelInstance } from "miragejs";
+  import Db from "miragejs/db";
+
+  type ModelInitializer<Data> = {
+    [K in keyof Data]: Data[K] extends Collection<infer M>
+      ? Collection<M> | M[]
+      : Data[K];
+  };
+
+  /**
+   * An interface to the Mirage ORM that allows for querying and creating records.
+   */
+  export default class Schema<Registry = Record<string, ModelInstance>> {
+    /** Mirage's in-memory database */
+    readonly db: Db;
+
+    /**
+     * Creates a model of the given type.
+     * @param modelName The type of model to instantiate
+     * @param data Optional initial values for model attributes/relationships
+     */
+    create<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<ModelInitializer<Init>>
+    >(
+      modelName: K,
+      data?: Data
+    ): Init & { [K in keyof Init & keyof Data]: Exclude<Init[K], undefined> };
+
+    /** Locates one or more existing models of the given type by ID(s). */
+    find<K extends keyof Registry>(type: K, id: string): Registry[K] | null;
+    find<K extends keyof Registry>(
+      type: K,
+      ids: string[]
+    ): Collection<Registry[K]>;
+
+    /** Locates an existing model of the given type by attribute value(s), if one exists. */
+    findBy<K extends keyof Registry>(
+      type: K,
+      attributes: Partial<Registry[K]>
+    ): Registry[K] | null;
+
+    /** Locates an existing model of the given type by attribute value(s), creating one if it doesn't exist. */
+    findOrCreateBy<K extends keyof Registry>(
+      type: K,
+      attributes: Partial<Registry[K]>
+    ): Registry[K];
+
+    /** Locates an existing model of the given type by attribute value(s), if one exists. */
+    where<K extends keyof Registry>(
+      type: K,
+      attributes: Partial<Registry[K]>
+    ): Collection<Registry[K]>;
+    where<K extends keyof Registry>(
+      type: K,
+      // tslint:disable-next-line:unified-signatures
+      test: (item: Registry[K]) => unknown
+    ): Collection<Registry[K]>;
+
+    /** Returns a collection of all known records of the given type */
+    all<K extends keyof Registry>(type: K): Collection<Registry[K]>;
+
+    /** Returns an empty collection of the given type */
+    none<K extends keyof Registry>(type: K): Collection<Registry[K]>;
+
+    /** Returns the first model instance found of the given type */
+    first<K extends keyof Registry>(type: K): Registry[K] | null;
+  }
+}
+
+declare module "miragejs/test-support/setup-mirage" {
+  /** Enables Mirage in the current test scope. */
+  export default function setupMirage(hooks: {
+    beforeEach: () => void;
+    afterEach: () => void;
```

Comment:
QUnit's hooks values technically pass `assert` to these functions, so we should do likewise here. It's rare that those should be used, but possible. It likely makes sense to import the relevant bits from the QUnit and/or ember-qunit types, so these can just stay in sync with them.

Reply:
Good suggestion. Thanks. `@types/qunit` added and the above definition is extended with `assert: Assert`.